### PR TITLE
Fixes with the full stack

### PR DESCRIPTION
**Pull request by harshil21**
Fixes:

- Servo Extensions
- State control fixes
- IMU.close() will not close the data queue (it will be done automatically)

### CODE REVIEW

File: airbrakes/mock/display.py [hunk 1]
```diff
@@ -60,8 +61,9 @@ def __init__(
         init(autoreset=True)  # Automatically reset colors after each print
         self._airbrakes = airbrakes
         self._start_time = start_time
-        self._mock = mock
-        self._verbose = verbose
+        self._args = args
+        # self._mock = mock
```
Comment (JacksonElia) [inline airbrakes/mock/display.py:65]: delete these comments

File: airbrakes/state.py [hunk 2]
```diff
@@ -146,10 +146,12 @@ def update(self):
...
         # if our prediction is overshooting our target altitude, extend the airbrakes
         if self.context.apogee_predictor.apogee > TARGET_ALTITUDE:
-            self.context.extend_airbrakes()
-        else:
-            # If our prediction is not overshooting, retract the airbrakes
+            if not self.airbrakes_extended:
+                self.context.extend_airbrakes()
+                self.airbrakes_extended = True
+        elif self.airbrakes_extended:
             self.context.retract_airbrakes()
```
Comment (JacksonElia) [inline airbrakes/state.py:153]: should check `if self.airbrakes_extended`

Comment (harshil21) [reply airbrakes/state.py:153]: done by ruff, it's correct

**Comment by harshil21** [comment]
Invalid fields now shown on display (and tested).

Untested code running `main.py -d`with real imu though

File: airbrakes/state.py (hunk 2 above)
Comment (JacksonElia) [reply airbrakes/state.py:153]: lol am i just blind

File: airbrakes/mock/display.py [hunk 3]
```diff
@@ -151,6 +156,13 @@ def _update_display(self, end_sim: Literal["natural", "interrupted"] | bool = Fa
         data_processor = self._airbrakes.data_processor
         apogee_predictor = self._airbrakes.apogee_predictor
 
+        if data_processor._last_data_packet:
```
Comment (JacksonElia) [inline airbrakes/mock/display.py:159]: fine

**Review by JacksonElia — APPROVED** [review]

**Pull request merged**